tests/Query/Filter/DueDateField.test.ts

```diff
@@ -50,6 +72,18 @@ describe('due date', () => {
         testTaskFilterForTaskWithDueDate(filter, '2022-04-25', false);
     });
 
+    it('by due date - on or before absolute range', () => {
```

Comment:
I had to spend some time reviewing these tests, and I have not reviewed them.

I think part of the challenge of reviewing them is that there is no statement of intent.

An example improvement would be to rename this test to:

```diff
- it('by due date - on or before absolute range', () => {
+ it('by due date - on or before absolute range should match all dates from the end of the range and earlier', () => {

Reply:
Thanks for clarifying the intent in https://github.com/obsidian-tasks-group/obsidian-tasks/pull/2186/commits/4c1dfc09c211653dafd9679d43aa30426567d0da